runtime/metricsview/executor_rewrite_approx_comparisons.go

```diff
@@ -100,15 +113,17 @@ func (e *Executor) rewriteApproxComparisonNode(a *AST, n *SelectNode) bool {
 		n.FromSelect.Limit = a.Root.Limit
 		n.FromSelect.Offset = a.Root.Offset
 
-		// ---- CTE Optimization ---- //
-		// make FromSelect a CTE
-		a.convertToCTE(n.FromSelect)
+		if cteOptimization {
+			// make FromSelect a CTE
+			a.convertToCTE(n.FromSelect)
 
-		// now change the JoinComparisonSelect WHERE clause to use selected dim values from CTE
-		for _, dim := range n.FromSelect.DimFields {
-			dimName := a.dialect.EscapeIdentifier(dim.Name)
-			dimExpr := "(" + dim.Expr + ")" // wrap in parentheses to handle expressions
-			n.JoinComparisonSelect.Where = n.JoinComparisonSelect.Where.and(fmt.Sprintf("%[1]s IS NULL OR %[1]s IN (SELECT %[2]q.%[3]s FROM %[2]q)", dimExpr, n.FromSelect.Alias, dimName), nil)
+			// now change the JoinComparisonSelect WHERE clause to use selected dim values from CTE
+			// this necessitates that order of dims in from and comparison select are same
+			for i := range n.FromSelect.DimFields {
+				dimName := a.dialect.EscapeIdentifier(n.FromSelect.DimFields[i].Name)
```

Comment:
Why access by `i` instead of `for _, dim := ...` and then using `dim.Name` like before?

Reply:
because I want to access corresponding dim from `ComparisonSelect` in the next line. So I want to use dim name and alias from base query for `IN (SELECT %[2]q.%[3]s FROM %[2]q)` and `dimExpr` from comparison to filter on using this condition and `IS NULL` condition. If dimName remains same across base and comparison then we may just use `for _, dim := range n.JoinComparisonSelect.DimFields` instead of  `n.FromSelect.DimFields` as you suggested.

Reply:
Yeah the dimension name is guaranteed to be the same – it is a fundamental guarantee that `SelectNode`s used in  joins must have the same `DimFields`